main.tf

```diff
@@ -15,8 +15,16 @@ locals {
       primary = var.ip_ranges.private_primary
       k8s = flatten([
         for k8s_ip_ranges in var.ip_ranges.private_k8s : [
-          { serial = index(var.ip_ranges.private_k8s, k8s_ip_ranges) + 1, cidr = k8s_ip_ranges.pods, name = format("private-k8spods%s-%s", "%s", var.name_suffix) },
-          { serial = index(var.ip_ranges.private_k8s, k8s_ip_ranges) + 1, cidr = k8s_ip_ranges.svcs, name = format("private-k8ssvcs%s-%s", "%s", var.name_suffix) }
+          {
+            cidr = k8s_ip_ranges.pods,
+            name = (k8s_ip_ranges.name != "") ? format("%s-pods", k8s_ip_ranges.name) : format("private-k8spods%s-%s",
+              (index(var.ip_ranges.private_k8s, k8s_ip_ranges) + 1) > 1 ? (index(var.ip_ranges.private_k8s, k8s_ip_ranges)) : "",
+          var.name_suffix) },
+          {
+            cidr = k8s_ip_ranges.svcs,
+            name = (k8s_ip_ranges.name != "") ? format("%s-svcs", k8s_ip_ranges.name) : format("private-k8ssvcs%s-%s",
+              (index(var.ip_ranges.private_k8s, k8s_ip_ranges) + 1) > 1 ? (index(var.ip_ranges.private_k8s, k8s_ip_ranges)) : "",
+          var.name_suffix) }
```

Comment:
If we are passing `name` inside a `object{}` structure, then the `name` will become a mandatory attribute - it's optional if it's inside a `map[]` structure. Hence, the condition `(k8s_ip_ranges.name != "") ?` will always be `true` as that attribute is mandatory.

It's actually good. I think then we an simplify the entire thing into this:

```terraform
{
  cidr = k8s_ip_ranges.svcs,
  name = format("private-k8ssvcs%s-%s", k8s_ip_ranges.name, var.name_suffix)
},
{
  cidr = k8s_ip_ranges.pods,
  name = format("private-k8spods%s-%s", k8s_ip_ranges.name, var.name_suffix)
}
```

We can then maintain backward-compatibility by passing in whatever `name` as needed. It won't break any format.

What do you say?

Reply:
@syedrakib to keep it backward compatible, I have kept the earlier implementation intact. however, if the user wants to use a custom name for secondary ranges to keep it concise and neat, it will use that name.

Reply:
My point was we can bring that entire name-construction (with index check greater than 1 or not etc etc) manually into our name attribute itself. While, at the same time, we maintain the existing format of range names as it is today. So no need to deviate from the current naming structure. That wins both the worlds.

Also the code remains cleaner with not so much if-else to read through and digest. 

Any particular reason why we would want to deviate from the existing naming pattern to something completely different?

Reply:
@syedrakib It's not to deviate but to give the name a logical meaning that's why I think it makes more sense to allow custom names.

For example, if i set the name as flights-secondary-range, the generated names for pods and svc would be `flights-secondary-range-pods` and `flights-secondary-range-svcs`, that sound more logical than `private-k8spodsflightssecondaryrange-nonprod` and `private-k8ssvcsflightssecondaryrange-nonprod`.

Now current implementation allows for both for the compatibility, we can still keep generating the names in this manner if the user doesn't provide it, however allowing him to logical name based on the use case. Even though we have been following this convention of `private-k8spods`, but we should always improvise the approach and not keep it rigid.

Reply:
Giving an option to have a custom name is welcomed. As there might be scenarios where it is needed. Keeping the recent `nginx-ingress` as an example, where we don't want their `ingress-nginx` name to stick around.

Reply:
Thanks for the elaboration @shardulsrivastava . It adds some context.

In that case, I would like to bring in some points.

We should not get rid of the name-suffix from the custom name. We are maintaining that approach for all our resource names across all of IAC and that suffix really helps catch silly errors early. It forces engineers to check if they are really referring to the correct resources for the intended environments. For example, it helped catch many such mistakes when people were writing kustomize files for instance. So we should keep the `var.name_suffix` in the custom names as well.

Also, now we don't need the `+ 1 > 1` here as we are dropping the concept of `serial` altogether (which was **1-based** - now, everything is back to **0-based**). So it's fine.

That being said, I propose to refactor these lines a little bit for the sake of readability & uniformity. 

```terraform
{
  cidr = k8s_ip_ranges.pods,
  name = (k8s_ip_ranges.name != "") ? (
    format("%s-pods-%s", k8s_ip_ranges.name, var.name_suffix)
  ) : (
    format(
      "private-k8spods%s-%s",
      index(var.ip_ranges.private_k8s, k8s_ip_ranges) > 0 ? index(var.ip_ranges.private_k8s, k8s_ip_ranges) : "",
      var.name_suffix
    )
  )
},
{
  cidr = k8s_ip_ranges.svcs,
  name = (k8s_ip_ranges.name != "") ? (
    format("%s-svcs-%s", k8s_ip_ranges.name, var.name_suffix)
  ) : (
    format(
      "private-k8ssvcs%s-%s",
      index(var.ip_ranges.private_k8s, k8s_ip_ranges) > 0 ? index(var.ip_ranges.private_k8s, k8s_ip_ranges) : "",
      var.name_suffix
    )
  )
}
```

It adds a few more number of lines for sure. But it creates an easier visual pattern to say which logic is ending where and which one is beginning where.

Let me know what you think.

Reply:
@syedrakib I agree with adding the `name_suffix`.
I  have updated the condition block accordingly.

Reply:
Great :)